app/src/main/java/org/mozilla/fenix/tabstray/CancelDownloadFragment.kt

```diff
@@ -0,0 +1,104 @@
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetDialog
+import org.mozilla.fenix.R
+import org.mozilla.fenix.databinding.FragmentCancelDownloadBinding
+import org.mozilla.fenix.ext.runIfFragmentIsAttached
+
+class CancelDownloadFragment : AppCompatDialogFragment() {
+
+    private var _binding: FragmentCancelDownloadBinding? = null
+    private val binding get() = _binding!!
+    private val args by navArgs<CancelDownloadFragmentArgs>()
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog =
+        BottomSheetDialog(requireContext(), this.theme).apply {
+            setOnShowListener {
+                val bottomSheet =
+                    findViewById<View>(com.google.android.material.R.id.design_bottom_sheet) as FrameLayout
+                val behavior = BottomSheetBehavior.from(bottomSheet)
+                behavior.state = BottomSheetBehavior.STATE_EXPANDED
+            }
+        }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentCancelDownloadBinding.inflate(inflater, container, false)
+
+        binding.cancelDownloadMessage.text = String.format(
+            binding.root.context.getString(
+                R.string.cancel_active_download_warning_content_description
+            ),
+            args.downloadCount
+        )
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.cancelDownloadAccept.setOnClickListener {
+            setFragmentResult(
+                RESULT_KEY,
+                Bundle().apply {
+                    putBoolean(ACCEPT_KEY, true)
+                    args.tabId?.let { putString(TAB_KEY, it) }
+                    args.source?.let { putString(SOURCE_KEY, it) }
+                    putBoolean(PRIVATE_KEY, args.private)
+                }
+            )
+            runIfFragmentIsAttached {
+                if (this.isVisible) {
+                    dismiss()
+                }
+                findNavController().popBackStack()
+            }
+        }
+
+        binding.cancelDownloadDeny.setOnClickListener {
+            dismiss()
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    companion object {
+        const val RESULT_KEY = "cancelDownloadFragmentResultKey"
+        const val ACCEPT_KEY = "cancelDownloadFragmentAcceptKey"
+        const val TAB_KEY = "cancelDownloadFragmentTabKey"
+        const val SOURCE_KEY = "cancelDownloadFragmentSourceKey"
+        const val PRIVATE_KEY = "cancelDownloadFragmentPrivateKey"
+    }
+}
+
+data class CancelDownloadFragmentArguments(
```

Comment:
I would recommend adding some kDoc for this data class and `Trigger`

Reply:
Would also consider renaming this so we don't have the `FragmentArguments` if it makes sense